core/arch/instrlist.h

```diff
@@ -116,6 +121,22 @@ instrlist_get_translation_target(instrlist_t *ilist);
 void
 instrlist_set_our_mangling(instrlist_t *ilist, bool ours);
 
+DR_API
+/**
+ * All future instructions inserted into \p ilist will be predicated
+ * with \p pred. This is a convenience routine to make it easy to have
+ * emitted code from internal dr components predicated.
+ *
```

Comment:
Mention the two restrictions: no cti (unless it's a clean call) and no writes to the flags.